test/tests/functional/pbs_soft_walltime.py

```diff
@@ -928,16 +928,33 @@ def test_soft_less_cput(self):
     dd if=/dev/zero of=/dev/null;
 done
 """
+	# If it is a cpuset mom, the cgroups hook relies on the periodic hook
+        # to update cput, so make the periodic hook run more often for the
+        # purpose of this test.
+        if self.mom.is_cpuset_mom():
+            attrs = {'freq':1}            
+            rv = self.server.manager(MGR_CMD_SET, HOOK, attrs, "pbs_cgroups")
+            self.assertEqual(rv,0)
+            # cause the change to take effect now
+            self.mom.restart()
```

Comment:
Why do you need to restart the mom to change the freq?  You set it on the server and the server sends  it to the mom.  Restarting  the mom shouldn't affect this.

Reply:
@bhroam : Unfortunately, the way the exechost_periodic hook has been implemented is that there'll be a delay when there's already one periodic hook queued up to execute at time X where X = <time previous hook executed and completed> + old 'freq' value. Only after that existing periodic hook executes, will a new periodic hook instance be queued to execute at a new 'freq' value. With pbs_cgroups hook having freq=120 default value, setting freq=1 won't actually take effect until that previous hook has awaken and completed execution. So setting 'freq=1' could actually get delayed up to the previous 'freq' value of 120 seconds + some leway time when mom is busy doing other tasks.
So the workaround is to either sleep 120+ seconds or just restart pbs_mom and the new 'freq' value immediately would take effect.  We decided it's best to just restart. 
Perhaps we should also file a ticket to change this behavior of exechost_periodic hook so that setting 'freq' value should be more instantaneous, on the mom side.

Reply:
Ahh, thanks for the explanation.  This makes sense.   Do we need to wait for the new freq value to be sent from the server before we restart?  Or will that be taken care of in the mom<->server handshake on startup?

Reply:
The new freq value is sent right away (as can be seen in the mom log), so no waiting is required.